staging/src/k8s.io/apimachinery/pkg/runtime/types_proto.go

```diff
@@ -27,43 +27,46 @@ type ProtobufMarshaller interface {
 // NestedMarshalTo allows a caller to avoid extra allocations during serialization of an Unknown
 // that will contain an object that implements ProtobufMarshaller.
 func (m *Unknown) NestedMarshalTo(data []byte, b ProtobufMarshaller, size uint64) (int, error) {
-	var i int
-	_ = i
-	var l int
-	_ = l
-	data[i] = 0xa
-	i++
-	i = encodeVarintGenerated(data, i, uint64(m.TypeMeta.Size()))
-	n1, err := m.TypeMeta.MarshalTo(data[i:])
-	if err != nil {
-		return 0, err
+	// Calculate the full size of the message.
+	msgSize := m.Size()
```

Comment:
@jennybuckley This size is *very slow* to compute, and it's been computed right before `NestedMarshalTo` is called, I think we should change that so that we don't double the call ...

Reply:
Or I don't know, I'm a little confused ...

Reply:
OK I got rid of this extra size call